test/source-sink-errors/mzcompose.py

```diff
@@ -0,0 +1,172 @@
+
+    def populate(self, c: Composition) -> None:
+        # Create a source and a sink
+        c.testdrive(
+            dedent(
+                """
+                > CREATE CONNECTION kafka_conn
+                  TO KAFKA (BROKER '${testdrive.kafka-addr}');
+
+                > CREATE CONNECTION IF NOT EXISTS csr_conn TO CONFLUENT SCHEMA REGISTRY (
+                  URL '${testdrive.schema-registry-url}'
+                  );
+
+                $ kafka-create-topic topic=source-topic
+
+                $ kafka-ingest topic=source-topic format=bytes
+                ABC
+
+                > CREATE SOURCE source1
+                  FROM KAFKA CONNECTION kafka_conn (TOPIC 'testdrive-source-topic-${testdrive.seed}')
+                  FORMAT BYTES
+                  ENVELOPE NONE
+                  /* WITH ( REMOTE 'storaged:2100' ) REMOTE causes the status source to be empty */
+
+                > CREATE SINK sink1 FROM source1
+                  INTO KAFKA CONNECTION kafka_conn (TOPIC 'testdrive-sink-topic-${testdrive.seed}')
+                  FORMAT AVRO USING CONFLUENT SCHEMA REGISTRY CONNECTION csr_conn
+                  ENVELOPE DEBEZIUM
+                  /* WITH ( REMOTE 'storaged:2100' ) REMOTE causes the status source to be empty */
+
+                # Wait for the source to start up
+                > SELECT COUNT(*) FROM source1
+                1
+                """
+            )
+        )
+
+    def assert_error(self, c: Composition, error: str) -> None:
+        # Validate that cluster2 continues to operate
+        c.testdrive(
+            dedent(
+                f"""
+                > SELECT status, error ~* '{error}'
+                  FROM mz_internal.mz_source_status
+                  WHERE name = 'source1'
+                stalled true
+
+                #> SELECT status, error ~* '{error}'
+                #  FROM mz_internal.mz_sink_status
+                #  WHERE name = 'sink1'
+                # stalled true
+                """
+            )
+        )
+
+    def assert_recovery(self, c: Composition) -> None:
+        c.testdrive(
+            dedent(
+                """
+                $ kafka-ingest topic=source-topic format=bytes
+                ABC
+
+                > SELECT COUNT(*) FROM source1;
+                2
+
+                #> SELECT status, error
+                #  FROM mz_internal.mz_source_status
+                #  WHERE name = 'source1'
+                #running <null>
+
+                #> SELECT status, error
+                #  FROM mz_internal.mz_sink_status
+                #  WHERE name = 'sink1'
+                #running <null>
+                """
+            )
+        )
+
+
+disruptions = [
+    Disruption(
+        name="delete-topic",
+        breakage=lambda c, seed: redpanda_topics(c, "delete", seed),
+        expected_error="UnknownTopicOrPartition",
+        fixage=None
+        # Re-creating the topic does not restart the source
```

Comment:
This is out of scope of this feature, but it seems that re-creating the topic does not cause ingestion to resume

Reply:
That sounds right, though I'm glad we can at least test that the topic-deleted error percolates up.